hipe4ml/TrainingAndTesting/plot_utils.py

```diff
@@ -0,0 +1,310 @@
+    scale = len(prediction[2]) / sum(hist)
+    err = np.sqrt(hist * scale) / scale
+    plt.errorbar(center, hist, yerr=err, fmt='o',
+                 c='b', label='Signal pdf Test Set')
+
+    # plt.gcf().subplots_adjust(left=0.14)
+    plt.xlabel('BDT output', fontsize=13, ha='right', position=(1, 20))
+    plt.ylabel(r'                                Counts (arb. units)', fontsize=13)
+    plt.legend(frameon=False, fontsize=12)
+    return res
+
+
+def plot_distr(sig_df, bkg_df, column=None, figsize=None, bins=50, log=False):
+    """Build a DataFrame and create two dataset for signal and bkg
+
+    Draw histogram of the DataFrame's series comparing the distribution
+    in `data1` to `data2`.
+
+    Input
+    -----------------------------------------
+    sig_df: signal candidates dataframe
+    bkg_df: background candidates dataframe
+    column: training columns
+    log: log scale plot
+
+    Output
+    -----------------------------------------
+    res: matplotlib object with the distributions of the features for
+    signal and background
+
+    """
+
+    data1 = bkg_df
+    data2 = sig_df
+
+    if column is not None:
+        if not isinstance(column, (list, np.ndarray, Index)):
+            column = [column]
+        data1 = data1[column]
+        data2 = data2[column]
+
+    if figsize is None:
+        figsize = [20, 15]
+
+    res = plt.figure()
+    axes = data1.hist(column=column, color='tab:blue', alpha=0.5, bins=bins, figsize=figsize,
+                      label='Background', density=True, grid=False, log=log)
+    axes = axes.flatten()
+    axes = axes[:len(column)]
+    data2.hist(ax=axes, column=column, color='tab:orange', alpha=0.5, bins=bins, label='Signal',
+               density=True, grid=False, log=log)[0].legend()
+    for axs in axes:
+        axs.set_ylabel('Counts (arb. units)')
+    return res
+
+
+def plot_corr(sig_df, bkg_df, columns, **kwds):
+    """Calculate pairwise correlation between features.
+
+    Input
+    -----------------------------------------------
+    sig_df: signal candidates dataframe
+    bkg_df: background candidates dataframe
+    column: training columns
+    **kwds: extra arguments are passed on to DataFrame.corr()
+
+    Output
+    ------------------------------------------------
+    fig: matplotlib object with the correlations between the
+    features for signal and background
+
+    """
+
+    data_sig = sig_df[columns]
+    data_bkg = bkg_df[columns]
+
+    corrmat_sig = data_sig.corr(**kwds)
+    corrmat_bkg = data_bkg.corr(**kwds)
+
+    tit = r'$\mathrm{\ \ \ ALICE \ Simulation}$ Pb-Pb $\sqrt{s_{\mathrm{NN}}}$ = 5.02 TeV'
+    fig = plt.figure(figsize=(20, 10))
+    # plt.title(t,y=1.08,fontsize=16)
+    plt.suptitle(tit, fontsize=18, ha='center')
+    grid = ImageGrid(fig, 111, nrows_ncols=(1, 2), axes_pad=0.15, share_all=True,
+                     cbar_location='right', cbar_mode='single', cbar_size='7%', cbar_pad=0.15)
```

Comment:
Maybe this part is relevant only for the hyper-triton analysis and should be reworked.